### PR TITLE
Add Dead Letter target config

### PR DESCRIPTION
**Pull request by jverhoeks**
Add Dead letter function to catch failed executions of the lambda

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -146,4 +147,11 @@ resource "aws_lambda_function" "default" {
       security_group_ids = [aws_security_group.default[0].id]
     }
   }
+
+  dynamic "dead_letter_config" {
+    for_each = local.dead_letter_config
+    content {
+      target_arn = var.dead_letter_target_arn
+    }
+  }
```
Comment (svanharmelen) [inline main.tf:156]: These blocks are ordered alphabetically.

File: variables.tf [hunk 2]
```diff
@@ -145,3 +145,9 @@ variable "tags" {
   type        = map(string)
   description = "A mapping of tags to assign to the bucket"
 }
+
+variable "dead_letter_target_arn" {
+  description = "The ARN of an SNS topic or SQS queue to notify when an invocation fails, it's required to add a policy."
+  type        = string
+  default     = null
+}
```
Comment (svanharmelen) [inline variables.tf:153]: Please make sure end with a newline on the last line.

Comment (svanharmelen) [reply variables.tf:153]: All variables are ordered alphabetically.

File: variables.tf [hunk 3]
```diff
@@ -145,3 +145,9 @@ variable "tags" {
   type        = map(string)
   description = "A mapping of tags to assign to the bucket"
 }
+
+variable "dead_letter_target_arn" {
+  description = "The ARN of an SNS topic or SQS queue to notify when an invocation fails, it's required to add a policy."
```
Comment (svanharmelen) [inline variables.tf:150]: Let's try to keep it short and concise:
```suggestion
  description = "The ARN of an SNS topic or SQS queue to notify when an invocation fails"
```

Comment (jverhoeks) [reply variables.tf]: Removed the text.

File: variables.tf (hunk 2 above)
Comment (jverhoeks) [reply variables.tf]: Changed the order

File: main.tf [hunk 4]
```diff
@@ -122,6 +123,13 @@ resource "aws_lambda_function" "default" {
   timeout                        = var.timeout
   tags                           = var.tags
 
+  dynamic "dead_letter_config" {
+    for_each = local.dead_letter_config
+    content {
+      target_arn = var.dead_letter_target_arn
+    }
+  }
```
Comment (svanharmelen) [inline main.tf:131]: Please see the other dynamic block and try to follow the same style:
```suggestion
  dynamic "dead_letter_config" {
    for_each = local.dead_letter_config
    
    content {
      target_arn = var.dead_letter_target_arn
    }
  }
```

**Review by svanharmelen — COMMENTED** [review]
Almost there...

File: main.tf (hunk 4 above)
Comment (jverhoeks) [reply main.tf]: Updated :)

File: main.tf [hunk 5]
```diff
@@ -146,4 +155,5 @@ resource "aws_lambda_function" "default" {
       security_group_ids = [aws_security_group.default[0].id]
     }
   }
+
```
Comment (svanharmelen) [inline main.tf:158]:
```suggestion
```

**Review by svanharmelen — COMMENTED** [review]
Nice, last comment and then we should be good...

File: main.tf (hunk 5 above)
Comment (jverhoeks) [reply main.tf]: Now it's clean 🧹

**Review by svanharmelen — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**